File: 10-stacks-queues1/1-intro/main.java
// Stack and Queue Introduction - Java Example

import java.util.Stack;
import java.util.LinkedList;
import java.util.Queue;

public class StacksQueuesIntro {
    public static void main(String[] args) {
        Stack<Integer> stack = new Stack<>();
        stack.push(1);
        stack.push(2);
        System.out.println("Stack pop: " + stack.pop());

        Queue<Integer> queue = new LinkedList<>();
        queue.add(1);
        queue.add(2);
        System.out.println("Queue pop: " + queue.poll());
    }
}

File: 10-stacks-queues1/1-intro/main.cpp
// Stack and Queue Introduction - C++ Example

#include <iostream>
#include <stack>
#include <queue>

int main() {
    std::stack<int> stack;
    stack.push(1);
    stack.push(2);
    std::cout << "Stack pop: " << stack.top() << std::endl;
    stack.pop();

    std::queue<int> queue;
    queue.push(1);
    queue.push(2);
    std::cout << "Queue pop: " << queue.front() << std::endl;
    queue.pop();

    return 0;
}
